src/edu/csus/ecs/pc2/ui/QuickJudgePane.java

```diff
@@ -0,0 +1,893 @@
+            judgementComboBox = new JComboBox<Judgement>();
+            judgementComboBox.setPreferredSize(new java.awt.Dimension(201, 25));
+            judgementComboBox.setPreferredSize(new java.awt.Dimension(201, 25));
+
+        }
+        return judgementComboBox;
+    }
+
+    /**
+     * This method initializes judgementSubmittedButton
+     * 
+     * @return javax.swing.JButton
+     */
+    private JButton getJudgementSubmittedButton() {
+        if (judgementSubmittedButton == null) {
+            judgementSubmittedButton = new JButton();
+            judgementSubmittedButton.setText("Submit");
+            judgementSubmittedButton.addActionListener(new java.awt.event.ActionListener() {
+                public void actionPerformed(java.awt.event.ActionEvent e) {
+                    judgeSelectedRuns();
+                }
+            });
+        }
+        return judgementSubmittedButton;
+    }
+
+    protected void judgeSelectedRuns() {
+
+        String separator = System.getProperty("line.separator");
+
+        showMessage("");
+
+        int[] selectedIndexes = runListBox.getSelectedIndexes();
+
+        if (selectedIndexes.length == 0) {
+            showMessage("Please select run to be judged");
+            return;
+        }
+
+        int numSelected = selectedIndexes.length;
+
+        Judgement judgement = (Judgement) judgementComboBox.getSelectedItem();
+
+        int result = FrameUtilities.yesNoCancelDialog(this, "Are you sure you want to judge " + numSelected + " runs as " + separator + judgement, "Judge said runs?");
+        if (result != JOptionPane.YES_OPTION) {
+            showMessage("Ok - nothing changed");
+            return;
+        }
+
+        int numJudged = 0;
+
+        for (int idx = 0; idx < selectedIndexes.length; idx++) {
+            int judgementIndex = getNextJudgementIndex();
+            judgement = (Judgement) judgementComboBox.getItemAt(judgementIndex);
+            ElementId id = (ElementId) runListBox.getRowKey(selectedIndexes[idx]);
+
+            Object[] cells = runListBox.getRow(selectedIndexes[idx]);
+            String state = (String) cells[4];
+            if (state.startsWith(Constants.DEL_RUN_PREFIX)) {
+                state = state.replace(Constants.DEL_RUN_PREFIX, "");
+            }
+            if (state.equals(RunStates.NEW.toString()) || state.equals(RunStates.QUEUED_FOR_COMPUTER_JUDGEMENT.toString())) {
+
+                slog("Quick Judge run " + id + " with judgement '" + judgement + "' ");
+
+                boolean judgedAsSolved = judgementIndex == 2;
+
+                Run run = getContest().getRun(id);
+
+                // Create judgement and save it in pendingJudgements
+                JudgementRecord judgementRecord = new JudgementRecord(judgement.getElementId(), getContest().getClientId(), judgedAsSolved, false);
+                pendingJudgements.put(run.getElementId(), judgementRecord);
+                System.out.println("Stored/Pending " + run);
```

Comment:
Probably want to remove System.out.xxxx or bracket it with a test for debugging or some such.

Reply:
Will replace System.out with log statements.

Reply:
pushed fix.